### ISSUE TITLE
Fatal error: Ssl not available

### ISSUE THREAD

**Issue by Chris00** [ocaml/opam-publish#58]
The error `Fatal error: Ssl not available` should be caught and the user should be said what to do — or were to complain!  Otherwise, for first time users, this can be quite frustrating.

At the moment the tool does not work without declaring `CONDUIT_TLS=native`.  It may be a good idea to try both SSL implementations before failing.

**Comment by AltGr** [comment]
Yes; shouldn't this be done in `conduit` itself, or at list `github`, though ? `opam-publish` doesn't make any direct reference to anything ssl-related... (in other words: yes, it's not even clear to me where to complain! And we didn't have this error with older versions of the libs)

**Comment by dbuenzli** [comment]
I'm hitting that right now. 

TBH  I think `opam-publish` should be rewritten with callouts to `curl`. Through the time the whole dep chain it's relying on has been a constant source of diverse failures for me. 

```
topkg opam submit
Submitting _build/hmap.0.8.1
[hmap-0.8.1.tbz] http://erratique.ch/software/hmap/releases/hmap-0.8.1.tbz downloaded
Fatal error: Ssl not available
Backtrace:
  Raised at file "src/core/lwt.ml", line 2719, characters 24-25
  Called from file "src/unix/lwt_main.ml", line 33, characters 8-18
  Called from file "src/cmdliner_term.ml", line 27, characters 19-24
  Called from file "src/cmdliner.ml", line 27, characters 27-34
  Called from file "src/cmdliner.ml", line 106, characters 32-39
  Called from file "src/cmdliner.ml", line 136, characters 18-36
  Called from file "src/cmdliner.ml", line 251, characters 22-48
topkg: [ERROR] run ['opam-publish' 'submit' '--msg'
                   
               '/var/folders/g4/nl6g_ytj63bbn9d45_jhggxc0000gn/T/topkg-opam-submit-msg-0b7053'
                    '_build/hmap.0.8.1']: exited with 1
```

@Chris00's suggestion works (but I would have had absolutely no clue where to find this): 

```
CONDUIT_TLS=native topkg opam submit
```

**Comment by mmottl** [comment]
I've also just run into this.  @Chris00 workaround works for me, too.

**Comment by pveber** [comment]
Sadly, @Chris00's workaround doesn't work for me:
```
$CONDUIT_TLS=native opam publish submit ./biocaml.0.8.0
[0.8.0.tar.gz] http://github.com/biocaml/biocaml/archive/0.8.0.tar.gz downloaded
Fatal error: Tls not available
```
I'm probably missing some optional dep somewhere, but couldn't find which...

**Comment by Chris00** [comment]
@pveber Did you install the package `tls`?

**Comment by pveber** [comment]
Yes, version `0.8.0`. I tried reinstalling `opam-publish` from a fresh `opam` switch and the command above now works. I remember `tls` was installed on my former switch prior to installing `opam-publish`, maybe there actually is a missing dependency somewhere, preventing something to be recompiled? In any case, my immediate problem is solved, thanks a lot!

**Comment by jberdine** [comment]
I hit this just now. I don't think I had any brew or opam ssl or tls packages installed, so I installed a bunch. The step that seemed to do the trick was `opam install lwt_ssl` followed by a manual `opam reinstall opam-publish`.

**Comment by XVilka** [comment]
Still reproducible.

**Comment by mmottl** [comment]
I've just tried @jberdine 's approach by first installing `lwt_ssl` and then reinstalling `opam-publish`.  I then had to also remove the previous workaround `CONDUIT_TLS=native` before executing `opam bistro` (which also does a `publish`).  This seems to be working just fine for me now.

**Comment by jberdine** [comment]
FTR, still repros today.

**Comment by Silur** [comment]
this means that OpenSSL method is no longer supported? only ocaml-tls?

**Comment by rjbou** [comment]
fixed by #71
